coalahtml/coala_html.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License
+# for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import http.server
+import json
+import os
+import socketserver
+from subprocess import call
+import sys
+import webbrowser
+
+from coalahtml.helper import (
+    get_file, create_dir, copy_files,
+    get_args, build_file_graph, parse_file_dict)
+from coalahtml import Constants
+from coalib.coala_main import run_coala
+from coalib.output.JSONEncoder import create_json_encoder
+from coalib.output.printers.ListLogPrinter import ListLogPrinter
+
+
+def main():
+    arg_parser = get_args()
+    args = arg_parser.parse_args()
+
+    dir_path = create_dir(os.path.abspath(args.dir))
+
+    if len(list(os.walk(dir_path))):
+        copy_files(get_file(Constants.COALA_HTML_BASE), dir_path)
+
+    if args.noupdate is False:
+        log_printer = ListLogPrinter()
+        results, exitcode, file_dict = run_coala(
+            log_printer=log_printer, autoapply=False, arg_parser=arg_parser)
+
+        retval = {"results": results}
```

Comment:
Actually it can also be interpreted as the value that has been returned. Naming sucks. :pensive: